src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentSearchService.kt

```diff
@@ -22,51 +18,33 @@ class AnonymousElectorDocumentSearchService(
     private val eroService: EroService,
     private val anonymousElectorDocumentSummaryRepository: AnonymousElectorDocumentSummaryRepository,
     private val anonymousSearchSummaryMapper: AnonymousSearchSummaryMapper,
+    private val specificationBuilder: AnonymousElectorDocumentSummarySpecificationBuilder,
 ) {
 
     fun searchAnonymousElectorDocumentSummaries(
-        dto: AnonymousSearchCriteriaDto
+        criteria: AnonymousSearchCriteriaDto
     ): AnonymousSearchSummaryResults {
-        val gssCodes = eroService.lookupGssCodesForEro(dto.eroId)
-        with(getAedPagedSummaries(dto = dto, gssCodes = gssCodes)) {
-            return AnonymousSearchSummaryResults(
-                page = dto.page,
-                pageSize = dto.pageSize,
+        val gssCodes = eroService.lookupGssCodesForEro(criteria.eroId)
+
+        val pageRequest = buildPageRequest(
+            page = criteria.page,
+            pageSize = criteria.pageSize
+        )
+        val specification = specificationBuilder.buildSpecification(gssCodes, criteria)
+
+        val aedSummaries = anonymousElectorDocumentSummaryRepository.findAll(specification, pageRequest)
+            .map { anonymousSearchSummaryMapper.toAnonymousSearchSummaryDto(entity = it) }
+        return with(aedSummaries) {
+            AnonymousSearchSummaryResults(
+                page = if (totalPages > 0) criteria.page else totalPages,
```

Comment:
The tests highlighted a small bug here, in the case where the query returns zero results. What this service used to return in that case was a `AnonymousSearchSummaryResults` with fields:
```
page: 1
pageSize: 100
totalPages: 0
totalResults: 0
results: []
```
This didnt feel quite right (ie. saying it was page 1 of 0 pages). How can it be page 1 of 0 ?
So I added the if expression to fix it.